### PR TITLE
Update max_table_size_to_drop and max_partition_size_to_drop with config reload

### PR DESCRIPTION
**Pull request by GrigoryPervakov**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Improvement

Changelog entry (up to few sentences, required except for Non-significant/Documentation categories):

Move setMax(Table|Partition)SizeToDrop call into config updater to change these settings without a restart.

### CODE REVIEW

File: dbms/tests/integration/test_reload_max_table_size_to_drop/test.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    config = open(CONFIG_PATH, 'r')
+    config_lines = config.readlines()
+    config.close()
+    config_lines = map(lambda line: line.replace("<max_table_size_to_drop>1", "<max_table_size_to_drop>1000000"),
+                       config_lines)
+    config = open(CONFIG_PATH, 'w')
+    config.writelines(config_lines)
+    config.close()
+
+    time.sleep(5)  # wait for config reload
```
Comment (Alex-Burmak) [inline dbms/tests/integration/test_reload_max_table_size_to_drop/test.py]: You can replace this sleep with:
```suggestion
    node.query("SYSTEM RELOAD CONFIG")
```

File: dbms/tests/integration/test_reload_max_table_size_to_drop/test.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+SCRIPT_DIR = os.path.dirname(os.path.realpath(__file__))
+CONFIG_PATH = os.path.join(SCRIPT_DIR, './_instances/node/configs/config.xml')
+
+
+@pytest.fixture(scope="module")
+def start_cluster():
+    try:
+        cluster.start()
+        node.query("CREATE TABLE test(date Date, id UInt32) ENGINE = MergeTree(date, id, 8192)")
```
Comment (Alex-Burmak) [inline dbms/tests/integration/test_reload_max_table_size_to_drop/test.py]: It's better to not use deprecated method for creating MergeTree tables (see https://clickhouse.yandex/docs/en/operations/table_engines/mergetree/).  
```suggestion
        node.query("CREATE TABLE test(date Date, id UInt32) ENGINE = MergeTree() PARTITION BY date ORDER BY id")
```

**Review by Alex-Burmak — CHANGES_REQUESTED** [review]

File: dbms/programs/server/Server.cpp [hunk 3]
```diff
@@ -438,6 +438,13 @@ int Server::main(const std::vector<std::string> & /*args*/)
             buildLoggers(*config, logger());
             global_context->setClustersConfig(config);
             global_context->setMacros(std::make_unique<Macros>(*config, "macros"));
+
+            /// Setup protection to avoid accidental DROP for big tables (that are greater than 50 GB by default)
+            if (config->has("max_table_size_to_drop"))
+                global_context->setMaxTableSizeToDrop(config->getUInt64("max_table_size_to_drop"));
```
Comment (tavplubix) [inline dbms/programs/server/Server.cpp]: `max_table_size_to_drop` in `Context::setMaxTableSizeToDrop(...)` and `Context::checkTableCanBeDropped(...)` can be accessed from different threads, but it's not atomic. The same for `max_partition_size_to_drop`

File: dbms/tests/integration/test_reload_max_table_size_to_drop/test.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        node.query("CREATE TABLE test(date Date, id UInt32) ENGINE = MergeTree() PARTITION BY date ORDER BY id")
+        yield cluster
+    finally:
+        cluster.shutdown()
+
+
+def test_reload_max_table_size_to_drop(start_cluster):
+    node.query("INSERT INTO test VALUES (now(), 0)")
+
+    time.sleep(5)  # wait for data part commit
```
Comment (tavplubix) [inline dbms/tests/integration/test_reload_max_table_size_to_drop/test.py]: It's commited in INSERT query, sleep is not needed

**Review by tavplubix — CHANGES_REQUESTED** [review]

File: dbms/tests/integration/test_reload_max_table_size_to_drop/test.py (hunk 4 above)
Comment (GrigoryPervakov) [reply dbms/tests/integration/test_reload_max_table_size_to_drop/test.py]: Without some time between INSERT and WAIT table will be successfully dropped with any value in max_table_size_to_drop. Due to the table size calculation algorithm(only committed parts summed), I assumed that it become committed in some time after INSERT.

**Review by tavplubix — APPROVED** [review]

**Comment by tavplubix** [comment]
msan report looks not related to this changes

**Pull request merged**